tests/test_timefuncs.py

```diff
@@ -1035,3 +1035,30 @@ def test_timedelta_range() -> None:
 def test_dateoffset_freqstr() -> None:
     offset = DateOffset(minutes=10)
     check(assert_type(offset.freqstr, str), str)
+
+
+def timedelta64_and_arithmatic_operator() -> None:
+    s1 = pd.Series(data=pd.date_range("1/1/2020", "2/1/2020"))
+    s2 = pd.Series(data=pd.date_range("1/1/2021", "2/1/2021"))
+    s3 = s2 - s1
+    td = np.timedelta64(1, "M")
+    check(assert_type((s1 - td), TimestampSeries), np.timedelta64)
+    check(assert_type((s1 + td), pd.Series[Any]), np.timedelta64)
+    check(assert_type((s1 * td), TimedeltaSeries), np.timedelta64)
+    check(assert_type((s1 / td), pd.Series[float]), np.timedelta64)
```

Comment:
For the above 4 checks, I am surprised they got past `pytest`, because the ehecks should be of the form `check(assert_type((s1 - td), TimestampSeries), pd.Series, pd.Timestamp)` (for example)